packages/react-aria-components/src/Table.tsx

```diff
@@ -587,7 +589,7 @@ export const TableHeader =  /*#__PURE__*/ createBranchComponent(
 
     return (
       <THead
-        {...mergeProps(filterDOMProps(props as any), rowGroupProps, hoverProps)}
+        {...mergeProps(filterDOMProps(props as any, {propNames: tableHeaderPropNames}), rowGroupProps, hoverProps)}
```

Comment:
could you remove the prop from the filterDOMProps (it's slow and nothing else returns that)
you can just pass it directly to the element
`onScroll={props.onScroll}`